### ISSUE TITLE
Agents need follow Transit_Walk in their plan, however they get PT

### ISSUE THREAD

**Issue by SarahSie** [matsim-org/matsim-code-examples#199]
Hello,

I do simulation in PT and some of my agents cannot perform their “transit_walk”. Instead, the program suggested them to use public transport.Therefore, they get error in this line:

ExperimentalTransitRoute route = (ExperimentalTransitRoute) basicAgentDelegate.getCurrentLeg().getRoute();

Because the current plan element of agents' plan is  transit_walk, which is Generic Route, However, the program send them to this part of the program to use PT.

It would be nice if someone could help me with that.
Thank you!

**Comment by SarahSie** [comment]
In the class “TransitAgentImpl”, for the method “getEnterTransitRoute”
I got following error

Caused by: java.lang.ClassCastException: org.matsim.core.population.routes.GenericRouteImpl cannot be cast to org.matsim.pt.routes.ExperimentalTransitRoute
	

The current plan element of the person is Transit_walk.
However, based on unknown reaso,n the agent supposes to use PT instead of performing trasit_walk


I will be happy if you help me to solve this error.

**Comment by kainagel** [comment]
Strange.  

(1) Could you please send that agent's plan?  (If it happens in an early iteration, there should be input_plans in that iteration's input directory.  If it happens in a later iteration, you may have to switch on that input plans are to be written in every iteration.)

(2) Could you please make the full logfile available?

Thanks

**Comment by SarahSie** [comment]
Thank you for your reply.
For the agents’ plan, I have run it for 500 iterations and it works well. 
However, when I use the withinday replanning, it faces error. This error happens for 78 agents. I should also mention that when I run the program just for one special agent alone, not with other agents, the agent can perform his plan. Nevertheless, when I run all agents together, then they face error.
The log file for the error is also attached.


[logfile.log.zip](https://github.com/matsim-org/matsim-code-examples/files/3567076/logfile.log.zip)

**Comment by kainagel** [comment]
Oh, I see.  If this uses within-day replanning for public transit, I can only say that this is not yet finally debugged.  (We are in between releases, after all, and 11.x did not have that at all.).  There is some version that seems to work in most cases, but it needs (1) the matsim development head, and (2) an explicitly set dependency to the newest release of a patched version of the sbb-raptor: https://github.com/vsp-gleich/matsim-sbb-extensions/releases .

In other words: If your life does not depend on it, I would advise to not use this yet, sorry.

**Comment by SahelHasti** [comment]
Thank you for your replay. I have a question regarding the transit_walk mode.
I know that the replanning suggests transit_walk for a link between
linkRefId="pt-8139" name="Zürich Oerlikon"
and 
linkRefId="pt-8088" name="Zürich Oerlikon"
and this is ok.
However, I have seen that when an agent cannot take his planned PT, then the replanning (router.calcRoute) suggest taking transit_walk between (for example) 
linkRefId="pt-13176" name="Zürich, Schumacherweg"
and
linkRefId="pt-13197" name="Zürich, Lerchenhalde"

Which is not a transit_walk, though it is somehow walking and changing the mode. Because the name and location of the stop stations are not the same.
I have also seen that replanning offers transit_walk from startLinkId=pt-2951 to endLinkId=822726. 
I will be happy if you help me to prevent that, the replanning suggests these kinds of transit_walk to the agents.
Therefore, I need that the agents remain at the station and only travel between two stations by pt, not transit_walk, or perform the transit_walk only when the name and location of the stations are the same.

**Comment by SahelHasti** [comment]
An update about the issue.
In the plan of the agent is to travel with Pt. However as I simulate a disruption then he cannot reach his vehicle and it goes to the withinday replanning by
List<Leg> newRoute = router.calcRoute(origin, destination, (time + reactTime), person);
In the new plan, the current plan element of the person is to travel from one bus stop to the next bus stop by transit_walk instead of the pt and from there, and then he can take another vehicle to continue his traveling.
Now the point is, the plan element of the agent before was traveling with pt and the person is still at the state of applying and waiting for the pt. However, after the withinday replan, the current state of the agent change to transit_walk.
So it would require that I can signal this to the mobility simulation somehow.
For making this signal, I need the help of someone who knows more about transit in the mobsim.

I will be thankful if I can receive your advice.
